src/dc-factory.ts

```diff
@@ -0,0 +1,164 @@
+import { findElementsForInit, isElementWithinLazyParent } from './dc-dom';
+import { DcBaseComponent } from 'src/dc-base-component'
+
+/**
+ * @typedef {string} ComponentState
+ * */
+
+/**
+ * @type {ComponentState}
+ */
+const STATE_NOT_INITED = 'not-inited';
```

Comment:
@Ofigelov component states could be moved into enum

Reply:
done